### PR TITLE
[V2] fix: Volume persistent-storage not csi error log

### LINKED ISSUE #1416: [V2] Fix "Volume persistent-storage not csi" misclassified error log line
**What happened**:
On pod creation reconciliation, we check the [claimToVolumeMap ](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/a9975e9ef8dfeab8c0656510bd8711d3d3a01c49/pkg/controller/common.go#L585) for AzDiskDriver volumes, but the pv is loaded to the map on its [update reconciliation](https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/a9975e9ef8dfeab8c0656510bd8711d3d3a01c49/pkg/controller/pv.go#L205) when it is bound which happens after pod is added. As a result for all volumes of the pod we end up returning a confusing error of "Volume persistent-storage not csi". I believe the reconciliation on pod update later fixes this misclassification but we need to change the workflow/error to remove the confusing error log. 

Here is where the log is printed: 

**What you expected to happen**:

**How to reproduce it**:

**Anything else we need to know?**:

**Environment**:
- CSI Driver version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by shlokshah-dev**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
[V2] Fix "Volume persistent-storage not csi" misclassified error log line

**Which issue(s) this PR fixes**:

Fixes #1416

**Requirements**:

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Review by landreasyan — APPROVED** [review]

File: pkg/controller/common.go [hunk 1]
```diff
@@ -565,8 +565,11 @@ func (c *SharedState) addPod(ctx context.Context, pod *v1.Pod, updateOption upda
 		}
 		namespacedClaimName := getQualifiedName(pod.Namespace, volume.PersistentVolumeClaim.ClaimName)
 		if _, ok := c.claimToVolumeMap.Load(namespacedClaimName); !ok {
-			w.Logger().V(5).Infof("Skipping Pod %s. Volume %s not csi. Driver: %+v", pod.Name, volume.Name, volume.CSI)
-			continue
+			// Log message if the Pod status is not Running
```
Comment (sunpa93) [inline pkg/controller/common.go:568]: Log message if the Pod status _is Running_?

File: pkg/controller/common.go [hunk 2]
```diff
@@ -565,8 +565,11 @@ func (c *SharedState) addPod(ctx context.Context, pod *v1.Pod, updateOption upda
...
+			if pod.Status.Phase == v1.PodRunning {
+				w.Logger().V(5).Infof("Skipping Pod %s. Volume %s not csi. Driver: %+v", pod.Name, volume.Name, volume.CSI)
+				continue
```
Comment (sunpa93) [inline pkg/controller/common.go:571]: We would need this `continue` to be outside of `if pod.Status.Phase == v1.PodRunning`.

File: pkg/controller/common.go (hunk 1 above)
Comment (shlokshah-dev) [reply pkg/controller/common.go]: Updated!

File: pkg/controller/common.go (hunk 2 above)
Comment (shlokshah-dev) [reply pkg/controller/common.go]: Updated

**Pull request merged**